src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/SpgProducerEeeCsvDataTests.cs

```diff
@@ -64,6 +65,192 @@ public async Task Execute_HappyPath_ReturnsProducerEeeWithSelectedYearAndObligat
+                Assert.Equal(20, result.Cat4Q2);
+                Assert.Equal(25, result.Cat5Q2);
+                Assert.Equal(30, result.Cat6Q2);
+                Assert.Equal(35, result.Cat7Q2);
+                Assert.Equal(40, result.Cat8Q2);
+                Assert.Equal(45, result.Cat9Q2);
+                Assert.Equal(50, result.Cat10Q2);
+                Assert.Equal(55, result.Cat11Q2);
+                Assert.Equal(60, result.Cat12Q2);
+                Assert.Equal(65, result.Cat13Q2);
+                Assert.Equal(70, result.Cat14Q2);
+                Assert.Equal(75, result.Cat15Q2);
+
+                Assert.Null(result.Cat1Q3);
+                Assert.Null(result.Cat1Q4);
+
+                Assert.Equal(1800, result.TotalTonnage);
+            }
+        }
+
+        [Fact]
+        public async Task Execute_HappyPath_ReturnsProducerEeeWithSelectedYearAndObligationType_As_B2B_WithAllCategories()
+        {
+            using (DatabaseWrapper db = new DatabaseWrapper())
+            {
+                //Arrange
+                ModelHelper helper = new ModelHelper(db.Model);
+                var scheme = helper.CreateScheme();
+                string approvalNumber = "WEE/TE0000ST/SCH";
+                scheme.ApprovalNumber = approvalNumber;
+                var memberUpload = helper.CreateSubmittedMemberUpload(scheme);
+                int complianceYear = DateTime.Now.Year;
+                memberUpload.ComplianceYear = complianceYear;
+                string prn1 = "PRN123";
+                string prn2 = "PRN456";
+                string b2bObligationType = EnumHelper.GetDisplayName(ObligationType.B2B);
+                string b2cObligationType = EnumHelper.GetDisplayName(ObligationType.B2C);
+
+                var producer1 = helper.CreateProducerAsCompany(memberUpload, prn1);
+                var producer2 = helper.CreateProducerAsCompany(memberUpload, prn2);
+
+                var dataReturnVersion1 = helper.CreateDataReturnVersion(scheme, complianceYear, 3);
+                var dataReturnVersion2 = helper.CreateDataReturnVersion(scheme, complianceYear, 4);
+
+                var categories = EnumHelper.GetValues(typeof(WeeeCategory));
+                var maxCategoryId = categories.Max(x => x.Key);
+                int quarter3Tonnage = 0;
+                int quarter4Tonnage = 0;
+
+                for (int i = 1; i <= maxCategoryId; i++)
+                {
+                    quarter3Tonnage = quarter3Tonnage + 10;
+                    quarter4Tonnage = quarter4Tonnage + 5;
+                    helper.CreateEeeOutputAmount(dataReturnVersion1, producer1.RegisteredProducer, b2bObligationType, i, quarter3Tonnage);
+                    helper.CreateEeeOutputAmount(dataReturnVersion2, producer1.RegisteredProducer, b2bObligationType, i, quarter4Tonnage);
+                }
+
+                helper.CreateEeeOutputAmount(dataReturnVersion1, producer2.RegisteredProducer, b2cObligationType, 2, 400);
```

Comment:
maybe add a comment to explain why adding this one, like in the previous test -- I missed the implication that it was b2c rather than b2b at first :)

Reply:
I think I've made it clearer with the comments I added